### DISCUSSION TITLE
clicking outside combobox keeps focus on input, how can i get it to blur

### DISCUSSION THREAD

**Discussion by badtant** [adobe/react-spectrum#3557] [Q&A] [answered]
Hi,

I'm currently trying out useCombobox and built this sandbox with a lot of help from the docs.
https://codesandbox.io/s/lucid-hugle-q8w953?file=/src/Popover.tsx
A problem I have is when the combobox is open and I click outside it. It closes but keeps focus on the input. How can I get it to also blur the input since I didn't click on the input?

I removed the restoreFocus prop from <FocusScope> (in Popover.tsx) but it still restores focus to the input.
Am I doing it wrong or is it a bug?

Thanks

**Comment by snowystinger** [comment, marked as answer]
UPDATE: See last few comments in this thread for the answer

Just click a second time to blur it. This is how native controls work too. See Select https://codesandbox.io/s/solitary-microservice-7p2vvv the first click will close it and the second will blur the control.

As for restoring, in a combobox, technically focus never left the input. It's virtually moving focus around in the popover. To verify this, try typing something in the input, then move up or down the menu in the popover using arrow keys. Note that the cursor remains in the input the entire time.

**Reply by badtant** [reply]
Yes, I understand the focus is still in the input. That's why I think it acts a little weird when I need to click outside twice even though I haven't moved into the popover with arrow keys. It's not what I'm used to using other comboboxes.

A few examples:
https://reach.tech/combobox/
https://ariakit.org/examples/combobox
https://jquense.github.io/react-widgets/docs/Combobox/
https://www.downshift-js.com/downshift

**Reply by snowystinger** [reply]
So my coworker pointed out that this doesn't happen in our React Spectrum Components https://react-spectrum.adobe.com/react-spectrum/ComboBox.html
This means this is doable with our hooks as we built our ComboBox on our hooks. I haven't looked yet to see the difference between our aria hook example and our component and I won't have time to look for a little bit. But this gives us a starting point at least.

**Reply by badtant** [reply]
hm, interesting.
one thing i see is that the popover doesn't seem to use FocusScope
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/overlays/src/Popover.tsx

**Reply by snowystinger** [reply]
Quickly tested that theory. Deleting the FocusScope in the docs code does not affect it. It was a good theory though.

**Reply by badtant** [reply]
Found a combination that seems to help
1. Remove (or set to false) restoreFocus from FocusScope props
2. Remove (or set to false) isDismissable from useOverlay props

Though I'm not quite sure if this has any other implications

**Reply by snowystinger** [reply]
That looks to be right. We don't use a restoring focus scope in our popover because focus is never actually in the popover. I'm not sure why the isDismissable false, but that's what we have in our React Spectrum Component as well. https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/combobox/src/ComboBox.tsx#L186

Good job sleuthing! I'll get the docs updated soon.

**Reply by badtant** [reply]
cool, then maybe DismissButton should be removed also?
